python/chemiscope/explore/__init__.py

```diff
@@ -124,39 +203,108 @@ def soap_kpca_featurize(frames, environments):
     .. _dscribe: https://singroup.github.io/dscribe/latest/
     .. _chemiscope-explore: https://chemiscope.org/docs/examples/6-explore.html
     """
-    # Check if dependencies were installed
-    try:
-        from pet_mad.explore import PETMADFeaturizer
-    except ImportError as e:
-        raise ImportError(
-            f"Required package not found: {e}. Please install the "
-            "dependencies with `pip install chemiscope[explore]`."
+
+    if "featurize" in kwargs:
+        if featurizer is not None:
+            raise ValueError(
+                "Both 'featurizer' and deprecated 'featurize' are provided"
+            )
+        featurizer = kwargs.pop("featurize")
+        warnings.warn(
+            "'featurize' was deprecated and renamed to 'featurizer'. Explicitly set "
+            "this parameter to silence this warning",
+            stacklevel=1,
         )
 
-    # Validate inputs
-    if featurize is not None and not callable(featurize):
-        raise TypeError("'featurize' must be a callable (function)")
     if properties is None:
         properties = {}
 
-    # Apply dimensionality reduction from the provided featurizer
-    if featurize is not None:
-        X_reduced = featurize(frames, environments)
-    else:
-        # Run default featurizer
-        featurizer = PETMADFeaturizer(
-            version="latest", device=device, batch_size=batch_size
+    extracted_properties = extract_properties(frames, environments)
+    merged_properties = {**extracted_properties, **properties}
```

Comment:
does this override things from the frame with whatever the user gave if they have the same name?